lib/Compiler.js

```diff
@@ -2,520 +2,524 @@
-};
 
-Compiler.prototype.newCompilation = function(params) {
-	var compilation = this.createCompilation();
-	compilation.fileTimestamps = this.fileTimestamps;
-	compilation.contextTimestamps = this.contextTimestamps;
-	compilation.name = this.name;
-	compilation.records = this.records;
-	compilation.compilationDependencies = params.compilationDependencies;
-	this.applyPlugins("this-compilation", compilation, params);
-	this.applyPlugins("compilation", compilation, params);
-	return compilation;
-};
-
-Compiler.prototype.createNormalModuleFactory = function() {
-	var normalModuleFactory = new NormalModuleFactory(this.options.context, this.resolvers, this.options.module || {});
-	this.applyPlugins("normal-module-factory", normalModuleFactory);
-	return normalModuleFactory;
-};
-
-Compiler.prototype.createContextModuleFactory = function() {
-	var contextModuleFactory = new ContextModuleFactory(this.resolvers, this.inputFileSystem);
-	this.applyPlugins("context-module-factory", contextModuleFactory);
-	return contextModuleFactory;
-};
+	createCompilation() {
+		return new Compilation(this);
+	}
 
-Compiler.prototype.newCompilationParams = function() {
-	var params = {
-		normalModuleFactory: this.createNormalModuleFactory(),
-		contextModuleFactory: this.createContextModuleFactory(),
-		compilationDependencies: []
-	};
-	return params;
-};
+	newCompilation(params) {
+		const compilation = this.createCompilation();
+		compilation.fileTimestamps = this.fileTimestamps;
+		compilation.contextTimestamps = this.contextTimestamps;
+		compilation.name = this.name;
+		compilation.records = this.records;
+		compilation.compilationDependencies = params.compilationDependencies;
+		this.applyPlugins("this-compilation", compilation, params);
+		this.applyPlugins("compilation", compilation, params);
+		return compilation;
+	}
 
-Compiler.prototype.compile = function(callback) {
-	var self = this;
-	var params = self.newCompilationParams();
-	self.applyPluginsAsync("before-compile", params, function(err) {
-		if(err) return callback(err);
+	createNormalModuleFactory() {
+		const normalModuleFactory = new NormalModuleFactory(this.options.context, this.resolvers, this.options.module || {});
+		this.applyPlugins("normal-module-factory", normalModuleFactory);
+		return normalModuleFactory;
+	}
 
-		self.applyPlugins("compile", params);
+	createContextModuleFactory() {
+		const contextModuleFactory = new ContextModuleFactory(this.resolvers, this.inputFileSystem);
+		this.applyPlugins("context-module-factory", contextModuleFactory);
+		return contextModuleFactory;
+	}
 
-		var compilation = self.newCompilation(params);
+	newCompilationParams() {
+		const params = {
+			normalModuleFactory: this.createNormalModuleFactory(),
+			contextModuleFactory: this.createContextModuleFactory(),
+			compilationDependencies: []
+		};
+		return params;
+	}
 
-		self.applyPluginsParallel("make", compilation, function(err) {
+	compile(callback) {
+		const self = this;
```

Comment:
`self` was only an alias for `this`. This is no longer needed with arrow functions